npm/vite-dev-server/client/initCypressTests.js

```diff
@@ -1,6 +1,42 @@
 // This file is merged in a <script type=module> into index.html
 // it will be used to load and kick start the selected spec
 
+// Fetch a dynamic import and re-try 3 times with a 2-second back-off.
+// We want to re-try these if they fail because sometimes (seemingly due to network issues)
+// the modules aren't available when we first request them.
+// See https://github.com/cypress-io/cypress/issues/25913
+async function importWithRetry (importFn) {
+  try {
+    return await importFn()
+  } catch (error) {
+    for (let i = 0; i < 3; i++) {
+      await new Promise((resolve) => setTimeout(resolve, 1000 * 2 ** i))
+
+      let url
+
+      try {
+        // Get request URL from error message from original import
+        url = new URL(
+          error.message
+          .replace('Failed to fetch dynamically imported module: ', '')
+          .trim(),
+        )
```

Comment:
Just out of curiosity, what sort of details do we get from this error message? Can we tell the difference between a module not being ready and the dev server being totally hosed (misconfigured, etc) so we don't unnecessarily hang the test on retries if it can't possibly succeed?